AG-16452 - Migrate benchmarks to in-example harness.

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16452

alantreadway committed:
AG-16452 - Enhance benchmark harness with version constraints and symlink resolution.

alantreadway committed:
AG-16452 - Reworked benchmarkHarness.ts to be symlink based.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Fix benchmark version restrictions.

alantreadway committed:
AG-16452 - Rework benchmark parameter display.

alantreadway committed:
AG-16452 - Fix benchmark append operation.

alantreadway committed:
AG-16452 - Create High Frequency benchmark examples.

alantreadway committed:
AG-16452 - Fix zoom handling in benchmarks.

alantreadway committed:
AG-16452 - Fix event handling in benchmarks.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Fix nx format of symlinks.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Verified working new benchmarks.

alantreadway committed:
AG-16452 - Fix review issues from iterative code review

- Fix cross-platform path handling in findSymlinkTargets() using path.relative()
- Remove stale hasBenchmarkConfig property from GeneratedContents interface
- Clarify performZoom documentation to match actual behavior (reuses chart)

lsjroberts commented on packages/ag-charts-website/src/content/docs/benchmarks/benchmarkUtils.ts:
```diff
@@ -0,0 +1,384 @@
+ * @param count - Number of toggles to perform
+ * @returns Elapsed time in milliseconds
+ */
+export async function performLegendToggle(
```
Should this click the legend item, using the `legendToggle()` function? Rather than directly toggling the series visibility.

alantreadway replied:
Good spot, adjusted to do that.

lsjroberts commented on packages/ag-charts-website/src/content/docs/high-frequency-data-test/_examples/high-freq-high-volume/benchmarkHarness.ts:
```diff
@@ -0,0 +1 @@
+../../../benchmarks/benchmarkHarness.ts
```
is this path correct? should it not be `../benchmarks/benchmarkHarness.ts`?

alantreadway replied:
I've verified that this is correct.

lsjroberts replied:
oh yeh, sorry, the path was truncated in the sidebar 😅

lsjroberts reviewed (approved):
LGTM, couple of questions

alantreadway committed:
Fix performLegendToggle to use DOM-based legend clicks

Address PR feedback: Update performLegendToggle to click legend items
via DOM events rather than directly updating series visibility through
the chart API. This matches the original Jest benchmark approach and
tests the full user interaction path including the accessibility layer.

Also adds await chart.waitForUpdate() before timing to ensure chart
stability, matching the pattern used in performZoom.
